WMIAdventure/backend/WMIAdventure_backend/IngameUsers/views.py

```diff
@@ -82,3 +85,55 @@ class UserDeckView(RetrieveAPIView):
     """
     serializer_class = UserDecksSerializer
     queryset = UserProfile.objects.all()
+
+
+class UserDeckView(RetrieveUpdateAPIView):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated, IsDeckOwner]
+
+    serializer_class = DeckSerializer
+    lookup_field = 'deck_number'
+
+    def get_object(self):
+        deck = get_object_or_404(self.get_queryset(), deck_number=self.kwargs['deck_number']).deck
+        self.check_object_permissions(self.request, deck)
+        return deck
+
+    def get_queryset(self):
+        user = get_object_or_404(User.objects.all(), pk=self.kwargs['pk'])
+        user_profile = user.userprofile
+        return user_profile.user_decks.all()
+
+    def _give_all_not_owned_cards_to_user(self, user_profile, request_data):
+        """
+        Gives cards to user if he doesn't own them.
+
+        TODO: REMOVE THIS IN THE FUTURE. For now all users should have all cards, this method will be removed when
+         there will be some way of gaining cards from battle or there will be some other way.
+
+        :raises django.http.Http404: If some of given cards does not exist.
+        """
+
+        cards = [
+            get_object_or_404(
+                Card.objects.all(),
+                info=request_data[f'card{i}']['id'],
+                level=request_data[f'card{i}']['level']
+            )
+            for i in range(1, 6)
+        ]
+        for card in cards:
+            UserCard.objects.get_or_create(
+                user_profile=user_profile,
+                card=card
+            )
+
+    def update(self, request, *args, **kwargs):
+        deck_to_update = self.get_object()
+        self._give_all_not_owned_cards_to_user(request.user.userprofile,

```

Comment:
Nie wiem czy callowanie tego za każdym razem jest dobrym pomysłem. To odpytuje bazę, dobrze byłoby jakbyśmy mogli jakoś sprawdzić łatwiej, czy user nie ma przypadkiem już wszystkich kart. Może sprawdzać samą ilość kart jakie są w bazie i ilośc kart jakie ma user?